File: StressTracker5001Server/DTOs/Board/BoardDto.cs
using StressTracker5001Server.DTOs.User;

namespace StressTracker5001Server.DTOs.Board
{
    public class BoardDto
    {
        public required int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public required UserDto Owner { get; set; }
        public required DateTime CreatedAt { get; set; }
        public required DateTime UpdatedAt { get; set; }
    }
}

File: StressTracker5001App/src/dto/board.dto.ts
import type { CardDto } from "./card.dto";
import type { ColumnDto } from "./column.dto";
import type { TagDto } from "./tag.dto";
import type { UserDto } from "./user.dto";

export interface BoardCreateDto {
  name: string;
  description: string;
}

export interface BoardUpdateDto {
  name: string;
  description: string;
}

export interface BoardDto {
  id: number;
  name: string;
  description: string;
  ownerId: number;
  owner: UserDto;
  createdAt: string;
  updatedAt: string;
}

export interface BoardDetailsDto extends BoardDto {
  columns: ColumnDto[];
  cards: CardDto[];
  tags: TagDto[];
}
